### PR TITLE
RR Exercise parser and evaluator

### PR DESCRIPTION
**Pull request by raulraja**
This PR is a POC and initial impl around #56 Parsing and evaluating capabilities.
The new backend scans the classpath and finds Sections, Categories and Exercises which are candidate  for inclusion in the webapp and allow for remote evaluation of responses based on the actual exercise compilation unit.
A few demo endpoints have been added as well as a sample class containing several exercises.

Required: @rafaparadela please review, thanks!
Optional : @FPerezP @juanpedromoreno

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -18,20 +16,26 @@ lazy val scalaExercisesServer = (project in file("scala-exercises-server")).sett
...
-    "org.postgresql"      % "postgresql"      % "9.3-1102-jdbc41",
-    "com.vmunier"         %% "play-scalajs-scripts" % "0.2.1",
-    "com.lihaoyi"         %% "upickle"        % "0.2.8",
-    "org.webjars"         %% "webjars-play"   % "2.3.0",
-    "org.webjars"         % "bootstrap"       % "3.2.0",
-    "org.webjars"         % "jquery"          % "2.1.1",
-    "org.webjars"         % "font-awesome"    % "4.1.0",
-    "com.tristanhunt"     %% "knockoff"       % "0.8.3"
+    specs2 % Test,
+    "com.typesafe.slick" %% "slick" % "3.0.0-RC1",
```
Comment (juanpedromoreno) [inline build.sbt]: Maybe we should upgrade to the latest version of slick: 3.1.0-M2

File: scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala [hunk 2]
```diff
@@ -10,45 +12,47 @@ import org.scalatest._
...
+class Extractors extends FlatSpec with Matchers with Exercises {
 
 
   /**
    * When you create a case class, it automatically can be used with pattern matching since it has an extractor:
    */
-  def extractorCaseClasses(solutions: Seq[Any]): Unit = {
+  def caseClasses(res0: String): ExerciseResult[Unit] = ExerciseRunner("Case Classes") {
+
+    println("Inside case classes args : " + res0)
```
Comment (juanpedromoreno) [inline scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala]: I think we don't want `println's` in the code, right?

**Comment by juanpedromoreno** [comment]
Just minor comments. Also I'm wondering if we are going to use specs2 or scalatest in this project, right now I see like a mix of both.

Anyway, code looks good and my congratulations for this big work!

File: build.sbt (hunk 1 above)
Comment (raulraja) [inline build.sbt]: Agreed though this is actually something Rafa is working on

File: scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala (hunk 2 above)
Comment (raulraja) [inline scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala]: This is actually just a sample class and should be replaced with the rest of the exercises in the stdlib package. I believe Rafa is porting all the Json data from v1 to this format and that point this class will dissappear

File: build.sbt [hunk 3]
```diff
@@ -18,20 +16,26 @@ lazy val scalaExercisesServer = (project in file("scala-exercises-server")).sett
...
+    "org.clapper" %% "classutil" % "1.0.5",
+    "com.toddfast.typeconverter" % "typeconverter" % "1.0",
+    "org.typelevel" %% "scalaz-specs2" % "0.3.0" % "test"
   )
- ).enablePlugins(PlayScala).
-  aggregate(clients.map(projectToRef): _*).
-  dependsOn(scalaExercisesSharedJvm)
+
+).enablePlugins(PlayScala).
+    aggregate(clients.map(projectToRef): _*).
```
Comment (FPerezP) [inline build.sbt]: Do we really want 4-space indentations?

File: build.sbt [hunk 4]
```diff
@@ -46,23 +50,36 @@ lazy val scalaExercisesClient = (project in file("scala-exercises-client")).sett
...
 // loads the jvm project at sbt startup
 onLoad in Global := (Command.process("project scalaExercisesServer", _: State)) compose (onLoad in Global).value
 
-// for Eclipse users
-EclipseKeys.skipParents in ThisBuild := false
\ No newline at end of file
+
+
+
+
```
Comment (FPerezP) [inline build.sbt]: We should avoid that big amount of empty lines.

File: scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala [hunk 5]
```diff
@@ -10,45 +12,47 @@ import org.scalatest._
  *
  *
  * object Twice {
- *   def apply(x: Int): Int = x * 2
- *   def unapply(z: Int): Option[Int] = if (z%2 == 0) Some(z/2) else None
+ * def apply(x: Int): Int = x * 2
+ * def unapply(z: Int): Option[Int] = if (z%2 == 0) Some(z/2) else None
```
Comment (FPerezP) [inline scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala]: No indentation here?

File: scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala (hunk 2 above)
Comment (FPerezP) [inline scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala]: IMHO, even if it's a sample class, we should avoid `printlns`. Why don't use a log.info or something like that? If we want to keep `println` we should create a ticket to make sure we remove this class once all json data has been moved to this format, just to not forget it.

**Comment by FPerezP** [comment]
Just minor comments about code style. It looks like you have something misconfigured in your IntelliJ. I don't mind if we use 2 or 4 space indentations, but we should be consistent. We are now using both styles. I'm being very picky but we should keep an eye to our code style too. Besides the indentation inconsistency, we have some extra white lines that should be removed too.

Other than that, LGTM.

**Comment by rafaparadela** [comment]
Nice job, @raulraja 

LGTM

File: scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala (hunk 2 above)
Comment (raulraja) [inline scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala]: Added #59 and assigned to @fedefernandez

File: scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala (hunk 5 above)
Comment (raulraja) [inline scala-exercises-content/src/main/scala/exercises/stdlib/Extractors.scala]: This is the result of automatic indentation. In any case I added #60 and assigned to @fedefernandez to configure Scalariform so we can have project wide standards regarding style and indentation and they are enforced in the SBT compile phase.

**Pull request merged**